Making the build path able to be modified using a buildPath parameter

adover opened the pull request:
To address the issue raised in https://github.com/InseeFrLab/keycloakify/issues/87. This PR begins the work of making the `build` directory configurable.

garronej commented on src/bin/build-keycloak-theme/build-keycloak-theme.ts:
```diff
@@ -14,11 +15,15 @@ type ParsedPackageJson = {
 const parsedPackageJson: ParsedPackageJson = require(pathJoin(reactProjectDirPath, "package.json"));
 
-export const keycloakThemeBuildingDirPath = pathJoin(reactProjectDirPath, "build_keycloak");
+const buildPath = argv("buildPath") ? String(argv("buildPath")) : "build";
```
the string "buildPath" shouldn't be repeated (for maintainability).  
Variables should be scoped as strictly as possible so that the code is easier 
to reason about.  

```ts
import {  assert } from "tsafe/assert";

//...

const buildPath= (()=>{

    const argName = "buildPath";

    const value = argv(argName);

    assert(value !== true, `expected --${argName} <path>, no <path> was provided`);

    return value || "build";

})();
```

garronej commented on src/bin/build-keycloak-theme/generateKeycloakThemeResources.ts:
```diff
@@ -109,6 +109,7 @@ export function generateKeycloakThemeResources(params: {
             "destDirPath": themeResourcesDirPath,
         });
 
+        // TODO:
```
TODO what?

garronej reviewed (commented):
Thank you very much for taking the time to submit your contribution!  

It you have the time to apply my suggested changes great, otherwise you gan grant me access to your fork so that I can apply them myself.  

An update in the readme explaining how to use the new parameter would be much appreciated.  

Grathfully,

adover commented:
Hi @garronej , thanks for your feedback. I have added you to the repository. There are still some fixes left to do, hence the `TODO` which I have added.

Webpack outside of CRA doesn't seem to use a `public` folder. We need to either a) build the `keycloak_static` to an assets directory and pre-set it, or b) reference the static files into the file system. I need to spend some time to understand your code to see what is happening there. Could you provide any pointers on how it works? Is it just a file replacement?

garronej commented:
Hi @adover,  
Thank you for your for your contribution.  
Two things before we move on with this:  
1) I just added a new feature, the ability to customize email template, this cause a tot of conflict with this PR.  
2) Have you considered creating a theme as a standalone project?

satanshiro commented on src/bin/build-keycloak-theme/argv.ts:
```diff
@@ -0,0 +1,9 @@
+
+    const value = process.argv.find(element => element.startsWith(`--${key}=`));
+
+    if (!value) return "";
```
sorry randomly got here why is this not returning false in contrast to line 2,

garronej replied:
@satanshiro,  
It should indeed?  
Are you reviewing this PR?  
It's not mine and have be stale for almost a year.  
If you submit another PR I'll review, merge and release promptly.  

Best

satanshiro replied:
oh wow I didn't notice the dates I was just wondering what work is being done, while I was checking to see if otp pages are available on keycloakify and it is. don't think I'll be needing this pr. thanks!

garronej commented:
Closed for inactivity, feel free to reopen.
